src/java.base/share/classes/java/util/Properties.java

```diff
@@ -903,10 +919,9 @@ private void store0(BufferedWriter bw, String comments, boolean escUnicode)
         if (comments != null) {
             writeComments(bw, comments);
         }
-        bw.write("#" + new Date().toString());
-        bw.newLine();
+        writeDateComment(bw);
         synchronized (this) {
-            for (Map.Entry<Object, Object> e : entrySet()) {
+            for (Map.Entry<Object, Object> e : new TreeMap<>(map).entrySet()) {
```

Comment:
Is this sorting intentionally added? It's not clear from issue description or PR description that order of properties should be changed too.
Anyway I think copying to array and then sorting should be faster, that creating TreeMap

Reply:
In case of reproducibility it should be at least ordered, i.e. keep original input order.